### PR TITLE
Switch to src layout and PDM

### PR DESCRIPTION
**Pull request by bsweger**
This PR switches the infrastructure repo to the Conda + PDM + src layout [described here]

The pulumi Python files have moved but haven't changed except for the import updates required for the change to src layout.

The next PR will follow-up with a test harness now that the repo is organized to support that.

Lastly, I updated the docs to reflect the new layout and made a few clarifications along the way.

### CODE REVIEW

**Comment by bsweger** [comment]
WIll need to get this one merged first (and rebase 😢 ) because that  branch contains the new workflow required by the new branch protections.

File: README.md [hunk 1]
```diff
@@ -1,63 +1,49 @@
...
-## Infrastructure as Code
+## Dev Setup
 
-Using a per-hub permissions model requires a set of tightly-integrated cloud resources. Using code to manage these resources (as opposed to creating them manually via a cloud provider's UI) provides several advantages:
+The code here uses a simple .yaml file that lists the cloud-enabled hubs. For each hub on the list, the Pulumi entry point invokes a Python function that provisions the required AWS resources.
 
-- Minimizes the potential for human error
-- Allows us to source control our cloud infrastructure management
-- Displays proposed changes to cloud infrastructure before they're applied
+If you're a Hubverse developer running Pulumi locally, you will need to setup the projectm and you will need the required Hubverse AWS credentials.
```
Comment (elray1) [inline README.md:47]:
```suggestion
If you're a Hubverse developer running Pulumi locally, you will need to setup the project and you will need the required Hubverse AWS credentials.
```

**Review by elray1 — COMMENTED** [review]
- I reviewed the PRs in the wrong order; saw you merged the other one and got those changes in here. now those checks are reporting that they can't find a file, but that file does seem to exist?
- I was able to create a conda env with python 3.11, activate it, and run `pdm sync`. it might be nice to provide some instructions about this?
- fixed a minor typo

**Comment by bsweger** [comment]
@elray1 Closing this one in favor of #31 

Sorry for all of the thrashing--this was one of those "should have been easy but ended up being quite annoying" items.

**Pull request closed without merging**